### ISSUE TITLE
change how we evaluate multi_target_models

### ISSUE THREAD

**Issue by louisfh** [kitzeslab/opensoundscape#716]
At the moment we set a `self.prediction_threshold = 0.5` for each CNN model object. This threshold choice seems arbitrary.

When we evaluate the performance of the model (by calling `multi_target_metrics`) on a validation dataset, we then report the precision, recall and f1 metrics using that prediction_threshold. How we evaluate the performance of the model depends on how we intend to use it, however I think our current setup could be improved.

e.g. If we want to use each class for top-down-listening, we might want to evaluate the performance of each class independently, using a metric like MAP

**Comment by sammlapp** [comment]
We changed the default metric to MAP (I think? or maybe area under ROC?) for multi target evaluation since precision/recall/F1 depend on threshold. I think the idea of `self.prediction_threshold` is that the user can modify that attribute if they want, but its just there to provide some sort of P/R/F1 during training. The user will evaluate the model afterwards.

**Comment by louisfh** [comment]
Part of the reason I think we should change it, is that by default we log the outputs of `multi_target_metrics` to wandb. I was watching a model train seeing the f1, precision and recall improve on wandb - however those metrics are arbitrary. Instead of reporting 3 arbitrary metrics for each class, I think `multi_target_metrics` should report average precision for each class (this is the same as area under the PR curve).

**Comment by sammlapp** [comment]
I agree! Logging F1, precision, and recall doesn't make sense for multi-target training. Logging average precision for each class sounds better.

**Comment by sammlapp** [comment]
I think the current behavior should be that it logs a bunch of metrics (including average precision, macro F1/precision/recall) for all scenarios. Is that what you see in WandB @louisfh? Do you suggest _removing_ F1/precision/recall during multi-target training?

**Comment by sammlapp** [comment]
Let's change the metrics that get printed during the training loop to focus on threshold-agnostic metrics. Let's also make sure to resolve #789

**Comment by sammlapp** [comment]
@louisfh currently
- training loop prints macro MAP and AU_ROC if model is multi-target (these metrics don't really make sense if you have a softmax layer ie single-target model)
- macro MAP and au_roc are logged to wandb and the metrics dictionary every epoch
- I'm adding per class map and au_roc to the metrics dict and wandb logging to resolve #789 

Does this resolve the issue? If not, comment with what you want the behavior to be

**Comment by sammlapp** [comment]
> OK that's fair enough. Is there a reason the threshold is 0.5? 

since the loss function is based on comparing scores in [0,1] to targets in {0,1}, 0.5 is the natural choice of threshold. (it corresponds to 0 on logit-scaled scores)
